### PR TITLE
Add React 18 example

### PR DESCRIPTION
**Pull request by vezaynk**

### CODE REVIEW

File: examples/index.tsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
 import { Titled } from '../src/Titled';
+import ReactDOM from 'react-dom';
```
Comment (IanVS) [inline examples/index.tsx]: This should be `react-dom/client`

File: examples/index.tsx [hunk 2]
```diff
@@ -44,4 +45,5 @@ export default class App extends React.Component {
   }
 }
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const root = ReactDOM.createRoot(document.getElementById('root'));
```
Comment (IanVS) [inline examples/index.tsx]: This will fail, since our `devDependencies` specifies `^16.8.0` for react, right?

File: examples/index.tsx (hunk 1 above)
Comment (vezaynk) [reply examples/index.tsx]: You are correct.

File: examples/index.tsx (hunk 2 above)
Comment (vezaynk) [reply examples/index.tsx]: That's true. Should I revert the change for compatability with 16 or upgrade the dev dependency to 18? I'm leaning towards the latter.

I can upgrade the #14 CR to run a matrix of React versions during CI to make sure tests pass with every (major) React version.

Comment (IanVS) [reply examples/index.tsx]: I like that idea.  Let's update the devDep, and at least run two react versions in CI, the oldest we support, and the latest 18.x.

**Comment by vezaynk** [comment]
@IanVS Done

**Pull request merged**